### ISSUE TITLE
hledger-ui: rfc: introduce navigation to immediate parent of a subaccount

### ISSUE THREAD

**Issue by ShrykeWindgrace** [simonmichael/hledger#1719]
Suppose we have a journal employing subaccounts, `hledger-ui --tree` and a subaccount selected.

Pressing "left arrow" or "h" key does nothing (for versions 1.21 and 1.23 as seen in irc chat on 24.09.2021).
I propose that these keys (and all their equivalents) move selection to the immediate parent of the selected subaccount.

Other possible behaviors in this scenario: collapse subaccounts - equivalent of sufficient number of "-" keypresses to show only the level of current subaccount's parent.

Priority - minor (for me, at least).

**Comment by la-wu** [comment]
What's the reason for using the 'left arrow' instead of introducing a new command (e.g lowercase `p`)? I don't personally use the cli-ui but I imagine it would be unfortunate to accidentally lose your position in the rendered tree list by holding the key for too long.

**Comment by simonmichael** [comment]
We discussed this a little in chat. The proposal is that on the accounts screen, if you are in tree mode and with the selection currently on a subaccount, pressing LEFT should move the selection to the immediate parent account. (And when selection is on a top-level account, it should do nothing.) @glguy contributed a patch to try it out.

I'm leaning against this:
- It complicates what LEFT does, making it more context-dependent. 
- For power users like us this is a familiar mechanic, but hledger-ui is also for less-technical folks. And for technical folks who don't want the controls taking up mental space.
- It would make adding a parent screen (such as a file chooser) more difficult.
- The benefit doesn't seem that great. Though I have only used this a little, let me know if the problem is more urgent in your usage @ShrykeWindgrace .

**Comment by ShrykeWindgrace** [comment]
@la-wu Indeed, come to think about it, there is no need to bind this action to the "left arrow" key, other meaningful mnemonic could be used - "p" being a good candidate, too. I guess my initial choice was inspired by `ncdu`/`nnn` and other TUI tools for folder navigation.

@simonmichael , per your points
- indeed
- I'd argue that the whole idea of using a cli/TUI software requires a poweruser to begin with =) Maybe we use different definitions of a "poweruser".
- Did not think about that, good point.
- As I said in irc, for me this is not urgent at all. If I understand that I can't live without it, I'm happy to patch the code and install from sources. I have no objections wrt putting this idea on hold if gains/efforts ratio seems to be low (in short term or in long term).

Cheers!
